universe/activities/activities.go

```diff
@@ -0,0 +1,249 @@
+package activities
+
+import (
+	"sort"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/momentum-xyz/ubercontroller/database"
+	"github.com/momentum-xyz/ubercontroller/types"
+	"github.com/momentum-xyz/ubercontroller/types/entry"
+	"github.com/momentum-xyz/ubercontroller/types/generic"
+	"github.com/momentum-xyz/ubercontroller/universe"
+	"github.com/momentum-xyz/ubercontroller/universe/activity"
+	"github.com/momentum-xyz/ubercontroller/utils/umid"
+)
+
+var _ universe.Activities = (*Activities)(nil)
+
+type Activities struct {
+	ctx        types.NodeContext
+	log        *zap.SugaredLogger
+	db         database.DB
+	activities *generic.SyncMap[umid.UMID, universe.Activity]
```

Comment:
hrm, same problem I had with the assets implementation. Probably an exponentially growing list, in memory where only the 'last' couple of entries are frequently used :/
So I would have kept it in DB, but I guess i'm too late for that again.